build.zig

```diff
@@ -187,9 +177,23 @@ pub fn build(b: *std.build.Builder) void {
     }
 
     {
+        const simulator_options = b.addOptions();
+
+        const StateMachine = enum { testing, accounting };
+        simulator_options.addOption(
+            StateMachine,
+            "state_machine",
+            b.option(
+                StateMachine,
+                "simulator-state-machine",
+                "State machine.",
+            ) orelse .accounting,
```

Comment:
As far as I know, there isn't a way in build.zig to say "this option applies only for 
`simulator` subcommand.".